pkg/kubelet/container/helpers.go

```diff
@@ -57,6 +57,10 @@ func ShouldContainerBeRestarted(container *api.Container, pod *api.Pod, podStatu
 	if status.State == ContainerStateRunning {
 		return false
 	}
+	// Always restart container in unknown state now
+	if status.State == ContainerStateUnknown {
+		return true
+	}
```

Comment:
No. If you do it after the restart policy checks, then the restart policy is honored, which isn't what we want here.